Structs/__init__.py

```diff
@@ -189,12 +189,25 @@ def convert_fstruct(
     
     def convert_arg(
         arg: Any,
-        idx: int
+        field: str,
     ) -> Any:
+        """
+        Helper function to handle FArray, FStruct and FScriptInterface arguments.
+
+        Recursively converts and FArray into an array.
+        Converts an FStruct arg into a named tuple.
+        Changes an FScriptInterface into the ObjectPointer of that FScriptInterface
+
+        Args:
+            arg: The Argument to handle.
+            field: The name of the field this struct was extracted from, used for error messages.
+        Returns:
+            The new argument.
+        """
         if isinstance(arg, unrealsdk.FArray):
-            return [convert_arg(a, i) for i, a in enumerate(arg)]
-        if isinstance(arg, unrealsdk.FStruct) and idx < len(fields):
-            return convert_fstruct(arg, fields[idx], type(defaults[idx]))
+            return [convert_arg(a, f"{field}[{idx}]") for idx, a in enumerate(arg)]
+        if isinstance(arg, unrealsdk.FStruct):
+            return convert_fstruct(arg, field, _all_structs[arg.structType.Name])
```

Comment:
Need to use `_get_struct_name`, sometimes we prepend to the base name. Kwargs have a different way of getting it, though they won't work with your nicer array error message format, which I think's good to keep. Should be able to use this method for both.

I don't think it's cleanly possible, but would be neat if the field error string (maybe rename it to `field_error` too) could concatenate across nested structs - `... to field 'OuterField[0].Inner.InnerInner'`

Reply:
renamed `field` to `field_error`, kwargs is now handled similar to args (while keeping check for key in fields. Could not find solution in the current state to keep the entire field chain, remains a todo for the future